src/particles/diagnostics/ReducedBeamCharacteristics.H

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Marco Garten, Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_REDUCED_BEAM_CHARACTERISTICS_H
+#define IMPACTX_REDUCED_BEAM_CHARACTERISTICS_H
+
+#include "particles/ImpactXParticleContainer.H"
+#include "ImpactX.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+#include <AMReX_GpuComplex.H>
+
+#include <cmath>
+
+using namespace amrex;
+
+namespace impactx::diagnostics
+{
+    /** Struct containing reduced beam characteristics data
+     */
+     struct ReducedBeamCharacteristics{
```

Comment:
One could also use a `std::unordered_map<std::string, amrex::ParticleReal>` instead. That way, the Python bindings are a `dict` and one does not need to bind another type.